packages/postgraphile-core/src/index.ts

```diff
@@ -265,22 +265,35 @@ export const getPostGraphileBuilder = async (
 
   let persistentMemoizeWithKey; // NOT null, otherwise it won't default correctly.
   let memoizeCache = {};
-
   if (readCache) {
-    const cacheString: string = await new Promise<string>((resolve, reject) => {
-      fs.readFile(readCache, "utf8", (err?: Error | null, data?: string) => {
-        if (err) {
-          reject(err);
-        } else {
-          resolve(data);
+    if (typeof readCache === "string") {
+      const cacheString: string = await new Promise<string>(
+        (resolve, reject) => {
+          fs.readFile(
+            readCache,
+            "utf8",
+            (err?: Error | null, data?: string) => {
+              if (err) {
+                reject(err);
+              } else {
+                resolve(data);
+              }
+            }
+          );
         }
-      });
-    });
-    try {
-      memoizeCache = JSON.parse(cacheString);
-    } catch (e) {
+      );
+      try {
+        memoizeCache = JSON.parse(cacheString);
+      } catch (e) {
+        throw new Error(
+          `Failed to parse cache file '${readCache}', perhaps it is corrupted? ${e}`
+        );
+      }
+    } else if (typeof readCache === "object") {
```

Comment:
```suggestion
    } else if (typeof readCache === "object" && !Array.isArray(readCache)) {
```

Reply:
I always forget how stupid js can be, thanks for the addition